backend.py

```diff
@@ -471,8 +471,10 @@ def rewriteNTPConf(root, ntp_servers):
     ntpsconf = open("%s/etc/chrony.conf" % root, 'w')
     for line in lines:
         ntpsconf.write(line)
-    for server in ntp_servers:
-        ntpsconf.write("server %s iburst\n" % server)
+
+    if ntp_servers:
```

Comment:
I haven't yet traced this to work out why exactly this occurs - this is to fix the issue quickly
I'll revisit this afterwards to identify the root cause but it's due the manual install not setting default values, this _should_ be caught by the calling code preconditions

Reply:
setTimeNTP is added to the tasks in backend.py:L111 with arguments "ntp-servers" and "ntp-config-method" from ans
In a manual install with no answerfile or answerfile generator, ans["ntp-servers"] is not populated by anything and therefore is None when the lookup fails
The introducing PR changed a check for the length of the ntp_servers list to a check for the ntp_config_method (dhcp/manual) [change](https://github.com/xenserver/host-installer/commit/e7f4c5abbf149de9c42d984b627573425812ad6a#diff-0922596903b115becd729d4bd53e9ad6e2d5a70ca42034806e650d49db0b5cccR482)
Previously there was no NTP setting which re-wrote the config (hence the need for the original change), so ntp_servers would always have been a list (when it was eventually used), but now it can be called without being populated which caused the failure.
The fix here (apart from adding the check in his PR) is to ensure ntp_servers is correctly populated when not using answerfiles (generators)